service-worker.js

```diff
@@ -26,17 +33,18 @@ self.addEventListener('notificationclick', function(event) {
   const notification = event.notification;
   notification.close();
 
-  const urlHash = '#/' + notification.data.topic;
+  const urlHash = '#/' + notification.data.xfrom;
```

Comment:
I think you found a bug, but this fix is incorrect. It won't work correctly for messages sent to group topics.